msal/src/main/java/com/microsoft/identity/client/GlobalSettingsConfiguration.java

```diff
@@ -0,0 +1,298 @@
+
+    /**
+     * The currently configured {@link HttpConfiguration} for use with the {@link PublicClientApplication}.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(HTTP)
+    private HttpConfiguration mHttpConfiguration;
+
+    /**
+     * The currently configured {@link LoggerConfiguration} for use with the {@link PublicClientApplication}.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(LOGGING)
+    private LoggerConfiguration mLoggerConfiguration;
+
+    /**
+     * Indicates whether the {@link PublicClientApplication} supports multiple clouds.  Automatic redirection to the cloud
+     * associated with the authenticated user.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(MULTIPLE_CLOUDS_SUPPORTED)
+    private Boolean mMultipleCloudsSupported;
+
+    /**
+     * Indicates whether the {@link PublicClientApplication} would like to leverage the broker if available.
+     * <p>
+     * The client must have registered the device.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(USE_BROKER)
+    private Boolean mUseBroker;
+
+    /**
+     * The {@link Environment} (Production, PPE) that the {@link PublicClientApplication} is registered in.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(ENVIRONMENT)
+    private Environment mEnvironment;
+
+    /**
+     * The minimum required broker protocol version number.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(REQUIRED_BROKER_PROTOCOL_VERSION)
+    private String mRequiredBrokerProtocolVersion;
+
+    /**
+     * The list of safe browsers.
+     */
+    @SerializedName(BROWSER_SAFE_LIST)
+    @Getter
+    @Accessors(prefix = "m")
+    private List<BrowserDescriptor> mBrowserSafeList;
+
+    /**
+     * The currently configured {@link TelemetryConfiguration} for the {@link PublicClientApplication}.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(TELEMETRY)
+    private TelemetryConfiguration mTelemetryConfiguration;
+
+    /**
+     * The currently configured {@link AccountMode} for the {@link PublicClientApplication}.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(ACCOUNT_MODE)
+    private AccountMode mAccountMode;
+
+    /**
+     * The currently configured capabilities for the {@link PublicClientApplication}.
+     */
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName(CLIENT_CAPABILITIES)
+    private String mClientCapabilities;
+
+    @Setter
+    @SerializedName(WEB_VIEW_ZOOM_CONTROLS_ENABLED)
+    private Boolean webViewZoomControlsEnabled;
+
+    @Setter
+    @SerializedName(WEB_VIEW_ZOOM_ENABLED)
+    private Boolean webViewZoomEnabled;
+
+    @Setter
+    @SerializedName(POWER_OPT_CHECK_FOR_NETWORK_REQUEST_ENABLED)
+    private Boolean powerOptCheckEnabled;
+
+    @SerializedName(HANDLE_TASKS_WITH_NULL_TASKAFFINITY)
+    private Boolean handleNullTaskAffinity;
```

Comment:
It seems like pretty much everything is becoming a Global Setting field. I don't think we want that. What's the reason behind things like `clientId` and `redirectUri` to be a global setting?

Reply:
Me and Shane started a discussion about this. I created a page in OneNote to decide which fields we would want to configure either in global, pca config, or the default msal config. We can set up some time today or tomorrow to discuss where we want to allow developers to configure fields.